File: FE/AL-homepage/src/types/classtime.ts
export interface Classtime {
  start_time: string // HH:MM:SS
  end_time: string // HH:MM:SS
  time_id: number
}

export interface CreateClasstimeRequest {
  start_time: string // HH:MM:SS
  end_time: string // HH:MM:SS
}

export interface UpdateClasstimeRequest {
  start_time?: string // HH:MM:SS
  end_time?: string // HH:MM:SS
}

export interface GetAllClasstimesParams {
  skip?: number
  limit?: number
}

File: BE/avalon/app/schemas/classtime.py
from pydantic import BaseModel
from typing import Optional
from datetime import time

class ClasstimeBase(BaseModel):
    start_time: time
    end_time: time

class ClasstimeCreate(ClasstimeBase):
    pass

class ClasstimeUpdate(BaseModel):
    start_time: Optional[time] = None
    end_time: Optional[time] = None

class Classtime(ClasstimeBase):
    time_id: int

    class Config:
        from_attributes = True